receiver/prometheusremotewritereceiver/receiver_test.go

```diff
@@ -406,17 +408,194 @@ func TestTranslateV2(t *testing.T) {
+				attrs.PutStr("service.instance.id", "host1")
+				attrs.PutStr("machine.type", "n1-standard-1")
+				attrs.PutStr("cloud.provider", "gcp")
+				attrs.PutStr("region", "us-central1")
+				attrs.PutStr("datacenter", "sdc")
+
+				sm := rm.ScopeMetrics().AppendEmpty()
+				sm.Scope().SetName("OpenTelemetry Collector")
+				sm.Scope().SetVersion("latest")
+
+				m := sm.Metrics().AppendEmpty()
+				m.SetName("normal_metric")
+				m.SetUnit("")
+				m.SetDescription("")
+
+				dp := m.SetEmptyGauge().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(1.0)
+				dp.SetTimestamp(pcommon.Timestamp(1 * int64(time.Millisecond)))
+				dp.Attributes().PutStr("d", "e")
+
+				return metrics
+			}(),
+		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			metrics, stats, err := prwReceiver.translateV2(ctx, tc.request)
+			// since we are using the interRequestCache to store values across requests, we need to clear it after each test, otherwise it will affect the next test
+			prwReceiver.interRequestCache = make(map[uint64]pmetric.ResourceMetrics)
+			metrics, _, err := prwReceiver.translateV2(ctx, tc.request)
 			if tc.expectError != "" {
 				assert.ErrorContains(t, err, tc.expectError)
 				return
 			}
 
 			assert.NoError(t, err)
 			assert.NoError(t, pmetrictest.CompareMetrics(tc.expectedMetrics, metrics))
-			assert.Equal(t, tc.expectedStats, stats)
 		})
 	}
 }
+
+type nonMutatingConsumer struct{}
+
+// Capabilities returns the base consumer capabilities.
+func (bc nonMutatingConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+type MockConsumer struct {
+	nonMutatingConsumer
+	mu         sync.Mutex
+	metrics    []pmetric.Metrics
+	dataPoints int
+}
+
+func (m *MockConsumer) ConsumeMetrics(_ context.Context, md pmetric.Metrics) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.metrics = append(m.metrics, md)
+	m.dataPoints += md.DataPointCount()
+	return nil
+}
+
+func TestTargetInfoWithMultipleRequests(t *testing.T) {
+	prwReceiver := setupMetricsReceiver(t)
+	mockConsumer := new(MockConsumer)
+	prwReceiver.nextConsumer = mockConsumer
+	w := httptest.NewRecorder()
+
+	firstRequest := &writev2.Request{
+		Symbols: []string{
+			"",
+			"job", "production/service_a", // 1, 2
+			"instance", "host1", // 3, 4
+			"machine_type", "n1-standard-1", // 5, 6
+			"cloud_provider", "gcp", // 7, 8
+			"region", "us-central1", // 9, 10
+		},
+		Timeseries: []writev2.TimeSeries{
+			{
+				Metadata:   writev2.Metadata{Type: writev2.Metadata_METRIC_TYPE_INFO},
+				LabelsRefs: []uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			},
+		},
+	}
+
+	buf := proto.NewBuffer(nil)
+	err := buf.Marshal(firstRequest)
+	assert.NoError(t, err)
+
+	req1 := httptest.NewRequest(http.MethodPost, "/api/v1/write", bytes.NewReader(buf.Bytes()))
+	req1.Header.Set("Content-Type", fmt.Sprintf("application/x-protobuf;proto=%s", promconfig.RemoteWriteProtoMsgV2))
+	req1.Header.Set("Content-Encoding", "snappy")
```

Comment:
Here is the snappy bug that I mentioned on slack. In the tests that you created.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/288c6bdedf283230b23d266a88059695cbc89a73/receiver/prometheusremotewritereceiver/receiver_test.go#L107-L114

I notice that you are compressing and sending the data through the request. But this just works in your test cases, because the body is empty, in the cases where we are sending `real data`, the code crashes, and the only way that I found to solve it, was sending the request without the snappy compression. 

Should we take a look into the `handlePRW` to deal with data that comes with that type of compression to avoid this behavior?

Reply:
If we have a bug unrelated to target_info, could we solve that first in a separate PR?

Reply:
> the only way that I found to solve it, was sending the request without the snappy compression.

This is not an option for us, [PRWv2 requires encoding messages with snappy](https://prometheus.io/docs/specs/prw/remote_write_spec_2_0/#:~:text=Protobuf%20Message%20MUST%20use%20binary%20Wire%20Format.%20Then%2C%20MUST%20be%20compressed%20with%20Google%E2%80%99s%20Snappy.%20Snappy%27s%20block%20format%20MUST%20be%20used%20%2D%2D%20the%20framed%20format%20MUST%20NOT%20be%20used.) :(

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38899

Fixed